test/Table_Tests/src/Formatting/Parse_Values_Spec.enso

```diff
@@ -395,22 +394,59 @@ spec =
             c1 = Column.from_vector "ints" ["0", "+0", "-0", "+1", "-1", "1", "000", "0010", "12345", Nothing]
             c2 = c1.parse type=Value_Type.Integer
             c2.name.should_equal c1.name
-            c2 . to_vector . should_equal [0, 0, 0, 1, -1, 1, Nothing, Nothing, 12345, Nothing]
+            c2 . to_vector . should_equal [0, 0, 0, 1, -1, 1, 0, 10, 12345, Nothing]
             c2.value_type.should_equal Value_Type.Integer
-            Problems.expect_warning Leading_Zeros c2
+            Problems.assume_no_problems c2
 
             c3 = c1.parse type=Value_Type.Integer format=(Data_Formatter.Value.with_number_formatting allow_leading_zeros=True)
             c3.to_vector . should_equal [0, 0, 0, 1, -1, 1, 0, 10, 12345, Nothing]
             Problems.assume_no_problems c3
 
+        Test.specify "should correctly parse integers in US formats" <|
+            cUS = Column.from_vector "ints" ["1", "000123", "-1234", "1234567", "123e456"]
+            pUS = cUS.parse type=Value_Type.Integer
+            pUS.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["123e456"]) pUS
+
+            cUS2 = Column.from_vector "ints" ["1", "000123", "-1,234", "1,234,567", "12,34,56"]
+            pUS2 = cUS2.parse type=Value_Type.Integer
+            pUS2.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["12,34,56"]) pUS2
+
+            cUS3 = Column.from_vector "ints" ["1", "(000,123)", "-1,234", "1,234,567", "12,34,56"]
+            pUS3 = cUS3.parse type=Value_Type.Integer
+            pUS3.to_vector . should_equal [1, -123, Nothing, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["-1,234", "12,34,56"]) pUS3
```

Comment:
I don't understand the difference between these two cases.

Why is in the first case `-1,234` parsed fine, but in the second case it fails to parse??? What is the difference?

Why is `(000,123)` parsed as `-123`? Where does the `-` come from? I'm confused. IS this some very weird US formatting?

Regardless of the behaviour of `(000,123)` which I assume has some explanation I just am not aware of, I don't understand how presence of `(000,123)` makes `-1,234` parse differently.

Reply:
Bracketted numbers are an accounting format for negative numbers (a stupid one but quite common).

The parser doesn't allow mixing of the formats so once it has settled that a column appears in bracket form it won't allow the minus sign.

Reply:
I see. I did not know that one. I think it would be good to add a note somewhere in our `Data_Formatter` or somewhere, about this bracketed format - I assume while well known to people that did accounting, many other kinds of users may be just as surprised as I was - so good to have some piece of explanation somewhere.

Reply:
What if the order is reversed? I first get `-1,234` and then **later** `(000,123)`. Will then the `-` format take precedence? How is the 'format affinity' determined?

I don't want to require tons of tests, but this is a pretty complicated and potentially brittle piece of logic so I think it would be good to add a test with reversed variant too. And maybe describe somewhere how this 'format affinity' is figured out. I guess once the parser does not do what the user expected, they may be interested in knowing what is happening.